recipes/ouster_sdk/all/conanfile.py

```diff
@@ -0,0 +1,223 @@
+    license = "BSD 3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ouster-lidar/ouster_example"
+    topics = ("ouster", "lidar", "driver", "hardware", "point cloud", "3d", "robotics", "automotive")
+
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "build_osf": [True, False],
+        "build_pcap": [True, False],
+        "build_viz": [True, False],
+        "eigen_max_align_bytes": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "build_osf": True,
+        "build_pcap": True,
+        "build_viz": False,
+        "eigen_max_align_bytes": False,
+    }
+    options_description = {
+        "build_osf": "Build Ouster OSF library.",
+        "build_pcap": "Build pcap utils.",
+        "build_viz": "Build Ouster visualizer.",
+        "eigen_max_align_bytes": "Force maximum alignment of Eigen data to 32 bytes.",
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "clang": "5",
+            "apple-clang": "10",
+            "msvc": "191",
+            "Visual Studio": "15",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if conan_version.major == 1:
+            # Turning off by default due to perpetually missing libtins binaries on CCI
+            self.options.build_pcap = False
+            self.options.build_osf = False
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        # Used in ouster/types.h
+        self.requires("eigen/3.4.0", transitive_headers=True)
+        # Used in ouster/sensor_http.h
+        self.requires("jsoncpp/1.9.5", transitive_headers=True, transitive_libs=True)
+        self.requires("spdlog/1.13.0")
+        self.requires("fmt/10.2.1")
+        self.requires("libcurl/[>=7.78 <9]")
+        # Replaces vendored optional-lite
+        self.requires("optional-lite/3.6.0", transitive_headers=True)
+
+        if self.options.build_pcap:
+            self.requires("libtins/4.5")
+
+        if self.options.build_osf:
+            # Used in fb_generated/*.h
+            self.requires("flatbuffers/24.3.7", transitive_headers=True)
+            self.requires("libpng/[>=1.6 <2]")
+            self.requires("zlib/[>=1.2.11 <2]", transitive_libs=True)
+
+        if self.options.build_viz:
+            self.requires("glad/0.1.36")
+            self.requires("CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS/3.4")
```

Comment:
@valgur what's that?

Reply:
Ooof. :grin: Thanks! Opened a PR: #23335.